broker/payment-receipt.go

```diff
@@ -127,24 +157,35 @@ func receiptInfoBuilder(policy models.Policy, transaction models.Transaction) do
 		Phone:      policy.Contractor.Phone,
 	}
 
-	expirationDate := lib.AddMonths(transaction.EffectiveDate, 12)
-	nextPayment := lib.AddMonths(transaction.EffectiveDate, 12)
+	expirationDate := policy.EndDate.AddDate(0, 0, -1)
+	nextPayment := "====="
+
+	tmpExpirationDate := lib.AddMonths(transaction.EffectiveDate, 12)
+	tmpNextPayment := lib.AddMonths(transaction.EffectiveDate, 12)
+
 	if policy.PaymentSplit == string(models.PaySplitMonthly) {
-		expirationDate = lib.AddMonths(transaction.EffectiveDate, 1).AddDate(0, 0, -1)
-		nextPayment = lib.AddMonths(transaction.EffectiveDate, 1)
+		tmpExpirationDate = lib.AddMonths(transaction.EffectiveDate, 1).AddDate(0, 0, -1)
+		tmpNextPayment = lib.AddMonths(transaction.EffectiveDate, 1)
+	}
+
+	if !tmpExpirationDate.After(expirationDate) {
+		expirationDate = tmpExpirationDate
+	}
+	if !tmpNextPayment.After(policy.EndDate) {
+		nextPayment = tmpNextPayment.Format(dateFormat)
 	}
 
 	payDate := "======="
 	if !transaction.PayDate.IsZero() {
-		payDate = transaction.PayDate.Format("02/01/2006")
+		payDate = transaction.PayDate.Format(dateFormat)
 	}
```

Comment:
Check if for the policy end date we have to subtract a day or not.

In the case of monthly policies we have a date for the expiry and the following day is the next payment day. But for yearly the dates are the same.

Reply:
I talked to Linda, and she told me that expiration date is effective date +1 month/year based on payment split